build-live.sh

```diff
@@ -0,0 +1,304 @@
+lxc exec $CONTAINER -- dpkg-divert --remove /sbin/initctl
+
+# clean up
+
+lxc exec $CONTAINER -- apt-get clean
+lxc exec $CONTAINER -- rm /var/lib/dbus/machine-id
+lxc exec $CONTAINER -- rm /etc/init/console.override
+lxc exec $CONTAINER -- rm /etc/init/tty1.override
+lxc exec $CONTAINER -- rm /etc/init/tty2.override
+lxc exec $CONTAINER -- rm /etc/init/tty3.override
+lxc exec $CONTAINER -- rm /etc/init/tty4.override
+# this will be recreated
+lxc exec $CONTAINER -- rm /etc/network/interfaces.d/eth0.cfg
+
+kversion=$(lxc exec $CONTAINER -- ls /boot/ | grep vmlinuz | grep -v efi | tail -1 | sed s@vmlinuz-@@)
+
+lxc file pull $CONTAINER/boot/vmlinuz-${kversion} ${CD}/${FS_DIR}/vmlinuz
+lxc file pull $CONTAINER/boot/initrd.img-${kversion} ${CD}/${FS_DIR}/initrd.img
+
+lxc exec $CONTAINER -- dpkg-query -W --showformat='${Package} ${Version}\n' | sort > ${CD}/${FS_DIR}/filesystem.manifest
+
+comm -2 -3 ${CD}/${FS_DIR}/filesystem.manifest ${CD}/${FS_DIR}/filesystem-desktop.manifest > ${CD}/${FS_DIR}/filesystem-remove.manifest
+
+lxc stop $CONTAINER
+
+cat > ${CD}/boot/grub/grub.cfg << EOF
+if loadfont /boot/grub/font.pf2 ; then
+    set gfxmode=auto
+    insmod efi_gop
+    insmod efi_uga
+    insmod gfxterm
+    terminal_output gfxterm
+fi
+
+set default="0"
+set timeout=10
+
+menuentry "Automated Install" {
+  linux /casper/vmlinuz boot=casper only-ubiquity automatic-ubiquity noprompt quiet splash file=/cdrom/preseed/posm-full.seed ip=frommedia
+  initrd /casper/initrd.img
+}
+
+menuentry "Install" {
+  linux /casper/vmlinuz boot=casper only-ubiquity noprompt quiet splash file=/cdrom/preseed/posm.seed ip=frommedia
+  initrd /casper/initrd.img
+}
+EOF
+
+# export from LXC in order to get correct UID/GIDs
+lxc publish $CONTAINER --alias $CONTAINER
+lxc image export $CONTAINER ${CONTAINER}.tar.gz
+lxc image delete $CONTAINER
+sudo rm -rf rootfs/
+sudo tar zxf ${CONTAINER}.tar.gz rootfs/
+rm ${CONTAINER}.tar.gz
+
+cat > ${CD}/${FS_DIR}/postinstall.sh <<EOF
+#!/bin/sh
+
+chroot /target mount -t proc proc /proc
+chroot /target mount -t sysfs sysfs /sys
+chroot /target mount -t securityfs securityfs /sys/kernel/security
+mount --bind /dev /target/dev
+mount --bind /run /target/run
+
+awk '{print \$1}' /cdrom/casper/filesystem-remove.manifest | xargs chroot /target apt-get purge -y
+chroot /target apt-get autoremove
+# TODO split wifi into package installation and configuration
+chroot /target /root/posm-build/kickstart/scripts/bootstrap.sh wifi captive
+EOF
+
+chmod +x ${CD}/${FS_DIR}/postinstall.sh
+
+cat > ${CD}/preseed/posm.seed <<EOF
+### Localization
+d-i debian-installer/locale string en_US.UTF-8
+
+### Clock and time zone setup
+d-i clock-setup/utc boolean true
+d-i time/zone string Etc/UTC
+d-i clock-setup/ntp boolean true
+d-i clock-setup/ntp-server string us.pool.ntp.org
+
+### Account setup
+d-i passwd/root-login boolean true
+d-i passwd/make-user boolean false
+d-i passwd/root-password-crypted password \$1\$foobar12\$1XX5fPEUg/6I1MhChN1ad1
```

Comment:
This should be configurable. Check if

```
d-i passwd/root-password password r00tme
d-i passwd/root-password-again password r00tme
```

override this when passed as boot arguments.